pkg/scheduler/podtask/port_mapping.go

```diff
@@ -2,168 +2,154 @@ package podtask
 
 import (
 	"fmt"
+	"sort"
 
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
-	log "github.com/golang/glog"
-	mesos "github.com/mesos/mesos-go/mesosproto"
 )
 
-type HostPortMappingType string
-
 const (
-	// maps a Container.HostPort to the same exact offered host port, ignores .HostPort = 0
-	HostPortMappingFixed HostPortMappingType = "fixed"
-	// same as HostPortMappingFixed, except that .HostPort of 0 are mapped to any port offered
-	HostPortMappingWildcard = "wildcard"
+	PortMappingLabelKey = "k8s.mesosphere.io/portMapping"
+	// FixedPorts represents a fixed port mapping strategy.
+	FixedPorts PortMapper = "fixed"
+	// WildcardPorts represents a fixed + wildcard port mapping strategy.
+	WildcardPorts PortMapper = "wildcard"
 )
 
-type HostPortMapper interface {
-	// abstracts the way that host ports are mapped to pod container ports
-	Generate(t *T, offer *mesos.Offer) ([]HostPortMapping, error)
-}
+// PortMapper represents a specifc port mapping strategy.
+type PortMapper string
 
-type HostPortMapping struct {
-	ContainerIdx int // index of the container in the pod spec
-	PortIdx      int // index of the port in a container's port spec
-	OfferPort    uint64
+// NewPortMapper returns an appropriate PortMapper for a given Pod.
+func NewPortMapper(pod *api.Pod) PortMapper {
+	filter := map[string]string{PortMappingLabelKey: "fixed"}
+	selector := labels.Set(filter).AsSelector()
+	if selector.Matches(labels.Set(pod.Labels)) {
+		return FixedPorts
+	}
+	return WildcardPorts
 }
 
-func (self HostPortMappingType) Generate(t *T, offer *mesos.Offer) ([]HostPortMapping, error) {
-	switch self {
-	case HostPortMappingWildcard:
-		return wildcardHostPortMapping(t, offer)
-	case HostPortMappingFixed:
+// PortMap computes PortMappings for the given Containers with the given offered port Ranges.
+func (m PortMapper) PortMap(offered Ranges, pod string, cs ...api.Container) ([]PortMapping, error) {
+	switch m {
+	case "wildcard":
```

Comment:
I left a comment here and I don't see it now. So I'm leaving another one.
Should reference the consts here